core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java

```diff
@@ -0,0 +1,279 @@
+                    yamlConfiguration = (Map<String, Object>) yaml.load(io);
+                } catch (IOException e) {
+                    throw new UncheckedIOException(e);
+                }
+            }
+        }
+
+        return parseConfiguration(yamlConfiguration);
+    }
+
+    public static Configuration loadPrecalculated() {
+        final File parent = Paths.get("").toAbsolutePath().toFile();
+        final File configFile = new LocalStorage(parent).duringExecution().temporary().file(SMART_TESTING_YML).getFile();
+
+        return loadConfigurationFromFile(configFile);
+    }
+
+    static Configuration loadConfigurationFromFile(File configFile) {
+        try(FileReader fileReader = new FileReader(configFile)) {
+            final Yaml yaml = new Yaml();
+            return yaml.loadAs(fileReader, Configuration.class);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load configuration from file " + configFile.getPath(), e);
+        }
+    }
+
+    public Path dump() {
+        final LocalStorageFileAction configFile = new LocalStorage(Paths.get("").toFile())
+            .duringExecution()
+            .temporary()
+            .file(Configuration.SMART_TESTING_YML);
+        try {
+            configFile.create();
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot create " + configFile.getPath() + " file", e);
+        }
+
+        try (FileWriter fileWriter = new FileWriter(configFile.getFile())) {
+            Yaml yaml = new Yaml();
+            yaml.dump(this, fileWriter);
+
+            return configFile.getPath();
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to store configuration in file " + configFile.getPath(), e);
+        }
+    }
+
+    // testing
+    public static Configuration load(Path path) {
+        try (InputStream io = Files.newInputStream(path)) {
+            final Yaml yaml = new Yaml();
+            Map<String, Object> yamlConfiguration = (Map<String, Object>) yaml.load(io);
+            return parseConfiguration(yamlConfiguration);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static Path getConfigurationFilePath(File[] files) {
+        if(files.length == 1) {
+            final File configFile = files[0];
+            logger.info("Using configuration from " + configFile.getName());
+            return configFile.toPath();
+        }
+
+        logger.warn("Found multiple config files with supported names: " + SMART_TESTING_YAML + ", "  + SMART_TESTING_YML);
+        logger.warn("Using configuration from " + SMART_TESTING_YML);
+
+        return Arrays.stream(files)
+            .filter(file -> file.getName().equals(SMART_TESTING_YML))
+            .map(File::toPath)
+            .findFirst()
+            .get();
+    }
+
+    private static Configuration parseConfiguration(Map<String, Object> yamlConfiguration) {
```

Comment:
This whole method is a little bit chaotic to my taste. I'm slightly afraid of adding additional props and also what it will look like when we support also configuration per strategy or when the size of parameters will grow. 
Have you considered my comment: https://github.com/arquillian/smart-testing/issues/113#issuecomment-328831162
Similar approach could be also useful for you - having one class "Configuration" and go through all variables stored in this class, based on the type and name read relative parameters from the file and store inside of the class instance. If the variable is something different than array/list/string/int/boolean and similar (in other words it is some additional object such as `Report`), then take this new class and do the same stuff also for this one... 
You know, some kind of generic engine that takes the yaml file and any Object and based on the structure read and fill corresponding values. 
Then, adding/removing parameters would be as easy as modifying the class structure.
Something similar I do for browser options in Drone: https://github.com/arquillian/arquillian-extension-drone/blob/master/drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/factory/CapabilitiesOptionsMapper.java
I think that you could find similar code also in Arquillian Core or other project...